### ISSUE TITLE
Writing myst documents programmatically

### ISSUE THREAD

**Issue by akhmerov** [executablebooks/MyST-Parser#238]
I'm looking into converting several large-ish repositories into executable books stack. Naturally I'd like to do as much as possible automatically, convert syntax, etc. Therefore an ability to manipulate and write a MyST document would be very useful. Unfortunately the only tool that can read myst—the parser, cannot write the result in myst format.

Would being able to write a myst document be within the roadmap of this project?

**Comment by chrisjsewell** [comment]
I think what you are after is what @mmcky has been working on: https://github.com/executablebooks/rst2myst

**Comment by mmcky** [comment]
hey @akhmerov -- been working on [sphinxcontrib-rst2myst](https://github.com/mmcky/sphinxcontrib-rst2myst) to support the bulk migration of the `quantecon` projects. I will move it into `executablebooks` once I meet @chrisjsewell standards `:-)`. 

It get's you most of the way there as it is now so please feel free to give it a go. Should just be a mater of adding the extension for a sphinx project in `conf.py` and running `make myst`. I'd love any feedback you may have. 

I am aiming to get full conversion including a `conf.py` passthrough. It would also be nice to auto-build for `jupyterbook` (i.e. `_config.yml` and `_toc.yml`) but haven't had time to work on that yet.

**Comment by chrisjsewell** [comment]
> I'm looking into converting several large-ish repositories into executable books stack.

I assume these are in rST, so have changed to issue name accordingly

**Comment by akhmerov** [comment]
> I assume these are in rST, so have changed to issue name accordingly

No, my source documents are in a different markdown flavor (mkdocs).

Parsing that isn't an issue, however writing myst from an ast is crucial.

**Comment by chrisjsewell** [comment]
Ah ok, so when you say parsing isn't an issue, how are you parsing it? What AST is being created? I take this is using mkdocs own render engine not sphinx.

**Comment by mmcky** [comment]
I guess `sphinxcontrib-rst2myst` isn't an entirely accurate name. It would work on any input -- so long as the input is parsed by `sphinx`. Perhaps I should rename it `sphinxcontrib-tomyst` but I didn't want to confuse it with `myst_parser` with name like `sphinxcontrib-myst` etc.

**Comment by akhmerov** [comment]
mkdocs uses python-markdown, and sphinxcontrib-tomyst addresses my needs rather well (with a bit of tinkering, I imagine).

**Comment by chrisjsewell** [comment]
Ok cool, well keep me posted

**Comment by chrisjsewell** [comment]
Heya, I'm going to close this as addressed by https://github.com/executablebooks/rst-to-myst/
